dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -245,6 +297,23 @@ public Contentlet findContentletForLanguage(long languageId,    Identifier conte
         return con;
     }
 
+    @CloseDBIfOpened
+    @Override
+    public List<Contentlet> findAllLangContentlets(final String identifier) {
+        final List<Language> languages = languageAPI.getLanguages();
+        final Identifier identifierObject = new Identifier();
+        identifierObject.setId(identifier);
+        return languages.stream().map(l -> {
+                    try {
+                        return findContentletForLanguage(l.getId(), identifierObject);
+                    } catch (Exception e) {
+                        Logger.error(this,"No working contentlet found for language");
+                    }
+                    return null;
+                }
+        ).filter(Objects::nonNull).collect(CollectionsUtils.toImmutableList());
+    }
+
```

Comment:
We have this method that is similar - I would like to avoid (where possible) hitting the index when managing a contentlet
```
	@Override
    protected Contentlet findContentletByIdentifierAnyLanguage(String identifier) throws DotDataException, DotSecurityException {
	    
	    // Looking content up this way can avoid any DB hits as these calls are all cached.
	    List<Language> langs = APILocator.getLanguageAPI().getLanguages();
	    for(Language l : langs) {
	        ContentletVersionInfo cvi = APILocator.getVersionableAPI().getContentletVersionInfo(identifier, l.getId());
	        if(cvi!=null && !cvi.isDeleted()) {
	            return find(cvi.getWorkingInode());
	        }
	    }
	    return null;

    }

```